graphbolt/src/index_select.cc

```diff
@@ -21,6 +22,25 @@ torch::Tensor IndexSelect(torch::Tensor input, torch::Tensor index) {
   return input.index({index.to(torch::kLong)});
 }
 
+torch::Tensor DiskIndexSelect(
+    std::string path, torch::Tensor index, torch::ScalarType dtype) {
```

Comment:
Why do we need dtype here?

Reply:
It is complex to parse the data type based on the meta data in the binary numpy file (so many types and needs a specialized function), a simple method is using high level python parse and transfer to C.

Reply:
When we expose OnDiskNpyArray to torch, we can take dtype in the constructor and DiskIndexSelect can take OnDiskNpyArray as its first argument instead of path.

Reply:
It can then use tensor.dtype to get it, so that getting the dtype separately is not necessary.